tests/influence/test_torch_differentiable.py

```diff
@@ -10,31 +10,27 @@
 import numpy as np
 import pytest
 
-from pydvl.utils import (
-    linear_regression_analytical_derivative_d2_theta,
-    linear_regression_analytical_derivative_d_theta,
-    linear_regression_analytical_derivative_d_x_d_theta,
+from .conftest import (
+    linear_derivative_analytical,
+    linear_hessian_analytical,
+    linear_mixed_second_derivative_analytical,
+    linear_model,
 )
 
-try:
-    import torch.nn.functional as F
+torch = pytest.importorskip("torch")
+import torch
+import torch.nn.functional as F
+from torch import nn
 
-    from pydvl.influence.frameworks import TorchTwiceDifferentiable
-    from pydvl.influence.model_wrappers import TorchLinearRegression
-except ImportError:
-    pass
+from pydvl.influence.frameworks.torch_differentiable import TorchTwiceDifferentiable
 
 
 class ModelTestSettings:
```

Comment:
We don't use classes to store parameters in the other test suites. Reasons:
1. Yuk!
2. `pytest.mark.parametrize` allows tuning for each test and is more explicit
3. Classes are data types, not data collections. If this were to be done (it won't), `ModelTestSettings` ought to be a dataclass or NamedTuple and an _instance_ would contain the values. Using class values for configuration is a pointless use of a class and unexpected for anyone seeing it.

Reply:
This was already here before so I suggest we create an issue and do this in a separate PR otherwise this won't be merged any time soon.

Reply:
I have removed the class. Not sure if here you also meant to write each input specifically instead of through itertools